app/src/main/java/com/eveningoutpost/dexdrip/GcmActivity.java

```diff
@@ -242,10 +242,11 @@ public void run() {
     }
 
     public synchronized static void syncBGReading(BgReading bgReading) {
+        Log.d(TAG, "syncBGReading called");
         if (JoH.ratelimit("gcm-bgs-batch", 15)) {
-            GcmActivity.sendMessage("bgs", bgReading.toJSON());
+            GcmActivity.sendMessage("bgs", bgReading.toJSON(true));
         } else {
-            PersistentStore.appendString("gcm-bgs-batch-queue", bgReading.toJSON(), "^");
+            PersistentStore.appendString("gcm-bgs-batch-queue", bgReading.toJSON(true), "^");
```

Comment:
I think batch queue entries also should be `false` as they are space sensitive and will come in a block preceded by a standard solo reading.